src/tools/r2rdump/R2RDump.cs

```diff
@@ -416,21 +449,22 @@ private int Run(string[] args)
             
             if (_help)
             {
-                Console.WriteLine(syntax.GetHelpText());
+                _writer.WriteLine(syntax.GetHelpText());
                 return 0;
             }
 
             if (_inputFilenames.Count == 0)
                 throw new ArgumentException("Input filename must be specified (--in <file>)");
 
-            // open output file
-            FileStream fileStream = null;
-            StreamWriter writer = null;
+            // open output stream
             if (_outputFilename != null)
             {
-                fileStream = new FileStream(_outputFilename, FileMode.Create, FileAccess.Write);
-                writer = new StreamWriter(fileStream);
-                Console.SetOut(writer);
+                FileStream fileStream = new FileStream(_outputFilename, FileMode.Create, FileAccess.Write);
+                _writer = new R2RWriter(fileStream);
+            }
+            else
+            {
+                _writer = new R2RWriter(Console.OpenStandardOutput());
```

Comment:
After my suggestion of changing the writer to just be a TextWriter, you should be able to just assign Console.Out to the variable instead of calling Console.OpenStandardOutput.